Core/src/test/java/com/rakuten/tech/mobile/perf/core/TelephonyUtilSpec.java

```diff
@@ -0,0 +1,90 @@
+package com.rakuten.tech.mobile.perf.core;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import android.content.Context;
+import android.telephony.TelephonyManager;
+import java.util.Locale;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class TelephonyUtilSpec {
+
+  @Mock private TelephonyManager telephonyManager;
+  @Mock private Context context;
+
+  @Before
+  public void initMocks() {
+    MockitoAnnotations.initMocks(this);
+    when(context.getSystemService(Context.TELEPHONY_SERVICE)).thenReturn(telephonyManager);
+  }
+
+  @Test
+  public void shouldReadCountryFromTelephonyManager() {
+    when(telephonyManager.getSimCountryIso()).thenReturn("TEST-SIM-COUNTRY");
+    assertThat(TelephonyUtil.getCountryCode(context)).isEqualTo("TEST-SIM-COUNTRY");
+  }
+
+  @Test
+  public void shouldFallbackToReadCountryFromLocaleIfCountryIsEmpty() {
+    when(telephonyManager.getSimCountryIso()).thenReturn("");
+    Locale.setDefault(new Locale("testLanguage", "TEST-LOCALE-COUNTRY", "testVariant"));
+
+    assertThat(TelephonyUtil.getCountryCode(context)).isEqualTo("TEST-LOCALE-COUNTRY");
+  }
+
+  @Test
+  public void shouldFallbackToReadCountryFromLocaleIfCountryIsNull() {
```

Comment:
i think the naming is a bit hard to understand
> should Fallback To Read Country From Locale

so far all good
> If Country Is Null

what?

looking at the code i think it should be something like
> if sim country is null

Reply:
Yeah, that's probably a better description.